frontend/src/__tests__/cypress/cypress/tests/e2e/pipelines/pipelines.cy.ts

```diff
@@ -0,0 +1,194 @@
+import {
+  createOpenShiftProject,
+  deleteOpenShiftProject,
+  applyOpenShiftYaml,
+} from '~/__tests__/cypress/cypress/utils/ocCommands';
+import { replacePlaceholdersInYaml } from '~/__tests__/cypress/cypress/utils/yaml_files';
+import { ADMIN_USER } from '~/__tests__/cypress/cypress/utils/e2eUsers';
+import { AWS_PIPELINES_BUCKET } from '~/__tests__/cypress/cypress/utils/s3Buckets';
+
+import { projectListPage, projectDetails } from '~/__tests__/cypress/cypress/pages/projects';
+import { pipelineImportModal } from '~/__tests__/cypress/cypress/pages/pipelines/pipelineImportModal';
+import { pipelinesGlobal } from '~/__tests__/cypress/cypress/pages/pipelines/pipelinesGlobal';
+import { pipelinesTable } from '~/__tests__/cypress/cypress/pages/pipelines/pipelinesTable';
+import { createRunPage } from '~/__tests__/cypress/cypress/pages/pipelines/createRunPage';
+import { pipelineRunDetails } from '~/__tests__/cypress/cypress/pages/pipelines/topology';
+
+const projectName = 'test-pipelines-prj';
+const dspaSecretName = 'dashboard-dspa-secret';
+const allAvailableProjectsText = 'All available projects';
+const testPipelineName = 'test-pipelines-pipeline';
+const testRunName = 'test-pipelines-run';
+
+describe('An admin user can import and run a pipeline', { testIsolation: false }, () => {
+  before(() => {
+    // Provision a Project
+    createOpenShiftProject(projectName).then((result) => {
+      expect(result.code).to.eq(
+        0,
+        `ERROR provisioning ${projectName} Project
+                                    stdout: ${result.stdout}
+                                    stderr: ${result.stderr}`,
+      );
+    });
+
+    // Create a pipeline compatible Data Connection
+    const dataConnectionReplacements = {
+      NAMESPACE: projectName,
+      AWS_ACCESS_KEY_ID: Buffer.from(AWS_PIPELINES_BUCKET.AWS_ACCESS_KEY_ID).toString('base64'),
+      AWS_DEFAULT_REGION: Buffer.from(AWS_PIPELINES_BUCKET.AWS_REGION).toString('base64'),
+      AWS_S3_BUCKET: Buffer.from(AWS_PIPELINES_BUCKET.BUCKET_NAME).toString('base64'),
+      AWS_S3_ENDPOINT: Buffer.from(AWS_PIPELINES_BUCKET.AWS_ENDPOINT).toString('base64'),
+      AWS_SECRET_ACCESS_KEY: Buffer.from(AWS_PIPELINES_BUCKET.AWS_SECRET_ACCESS_KEY).toString(
+        'base64',
+      ),
+    };
+    cy.fixture('resources/yaml/data_connection.yml').then((yamlContent) => {
+      const modifiedYamlContent = replacePlaceholdersInYaml(
+        yamlContent,
+        dataConnectionReplacements,
+      );
+      const tempFilePath = 'cypress/temp_data_connection.yaml';
+      applyOpenShiftYaml(modifiedYamlContent, tempFilePath).then((result) => {
+        expect(result.code).to.eq(
+          0,
+          `ERROR applying YAML content
+                                      stdout: ${result.stdout}
+                                      stderr: ${result.stderr}`,
+        );
+      });
+    });
```

Comment:
Fixtures / resource creation should be placed in reusable utility functions.

Reply:
Fixed in [6a949ea](https://github.com/opendatahub-io/odh-dashboard/pull/3028/commits/6a949ea4f0fbc70b412b2e9462af4a8873ad8d03)